Web Viewer Bookmarks

stefandanzl opened the pull request:
# I am submitting a new Community Plugin

## Repo URL


Link to my plugin: https://github.com/stefandanzl/webviewer-bookmarks

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

bnchiang96 commented:
Closing this PR in favor of a new submission with version 1.1.0, which addresses styling feedback by moving styles to styles.css and includes bug fixes for the modal.

stefandanzl commented:
> Closing this PR in favor of a new submission with version 1.1.0, which addresses styling feedback by moving styles to styles.css and includes bug fixes for the modal.

I'm confused - will my plugin be able to be added?
Or should i open a new PR with my latest 1.1.0 release?

joethei commented:
Why is this a plugin?
What's wrong with the default bookmarks plugin?

[# Obsidian Web Bookmarks](https://github.com/stefandanzl/webviewer-bookmarks/blob/97304380eb6bf765e981ddc0bee83b17c7c1a91f/README.md#L1)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

[name: `Open Web Viewer Bookmark: ${this.getDisplayName(](https://github.com/stefandanzl/webviewer-bookmarks/blob/97304380eb6bf765e981ddc0bee83b17c7c1a91f/main.ts#L45), [.setButtonText("Add Bookmark")](https://github.com/stefandanzl/webviewer-bookmarks/blob/97304380eb6bf765e981ddc0bee83b17c7c1a91f/main.ts#L151), [text: "Lucide Icons",](https://github.com/stefandanzl/webviewer-bookmarks/blob/97304380eb6bf765e981ddc0bee83b17c7c1a91f/main.ts#L176), [.setName("Show in Ribbon")](https://github.com/stefandanzl/webviewer-bookmarks/blob/97304380eb6bf765e981ddc0bee83b17c7c1a91f/main.ts#L254), [.setName("Create Command")](https://github.com/stefandanzl/webviewer-bookmarks/blob/97304380eb6bf765e981ddc0bee83b17c7c1a91f/main.ts#L265), [.setName("Open in New Tab")](https://github.com/stefandanzl/webviewer-bookmarks/blob/97304380eb6bf765e981ddc0bee83b17c7c1a91f/main.ts#L279), [.setName("Reload App")](https://github.com/stefandanzl/webviewer-bookmarks/blob/97304380eb6bf765e981ddc0bee83b17c7c1a91f/main.ts#L290)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[containerEl.createEl("h2", { text: "Web Viewer Bookmarks Settings" });](https://github.com/stefandanzl/webviewer-bookmarks/blob/97304380eb6bf765e981ddc0bee83b17c7c1a91f/main.ts#L145)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

stefandanzl commented:
This plugin is not to bookmark markdown files but website urls as ribbon icons and commands and as far as I can tell there is neither a core plugin or functionality nor a community plugin available for this.

I changed the first heading of Readme.md and added the mention of "community plugin" to further clarify the inofficial nature.

I changed all the settings strings to follow the sentence case style.

I removed the heading in the settings page.

stefandanzl commented:
Do I have to edit obsidian-releases.json again or will this be done automatically?

I published a new release with the changes

stefandanzl commented:
Hello, this plugin still needs approval, please let me know what to change or merge it.

Moyf commented:
> Hello, this plugin still needs approval, please let me know what to change or merge it.

I think the bot will regularly scan *each line* of the code content mentioned in the review, and only after all of them have been modified will it automatically add or change Labels.  
And it seems there's a command to skip the scanning and have a human review it again.

I feel a bit regretful—there doesn’t seem to be any documentation explaining the "bot/human review process" (or maybe I just couldn’t find it). I’ve seen several plugin authors waiting for updates until their PR were marked as `stale`.
